### DISCUSSION TITLE
Is there a way to move rules using the API?

### DISCUSSION THREAD

**Discussion by b-ryan** [firefly-iii/firefly-iii#5540] [Firefly III] [answered]
Hi there, I'm working on implementing the API code I mentioned in https://github.com/firefly-iii/firefly-iii/issues/5530#issuecomment-1011382297

I have functional code to create and delete rules via the API, but I'd also like to be able to re-arrange the rules using the API. The PUT endpoint for rules doesn't seem to have any effect. I also saw there is a `move-rule` route that the application uses, but it's not exposed as an API endpoint.

**Comment by JC5** [comment, marked as answer]
That's a bug, nice find. I'll push a fix to the develop branch, which will find its way into a new release on of these days. To move a rule, simply set the `order` field to the desired position.

**Reply by b-ryan** [reply]
thanks!
